modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/GridCacheOffheapManager.java

```diff
@@ -2522,7 +2587,7 @@ private Metas getOrAllocatePartitionMetas() throws IgniteCheckedException {
                             partMetastoreReuseListAllocated = true;
                         }
 
-                        if ((updateLogTreeRoot = io.getUpdateTreeRoot(pageAddr)) == 0) {
+                        if ((updateLogTreeRoot = io.getUpdateTreeRoot(pageAddr)) == 0 && preallocateUpdateLogTree) {
```

Comment:
why we do it again? At line 2541 we already allocated page for case preallocateUpdateLogTree

Reply:
sorry, my bad.